src/Symfony/Component/Lock/Tests/Store/ZookeeperStoreTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+/*
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ *
+ * @requires extension zookeeper
+ */
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\ZookeeperStore;
+use Throwable;
+use Zookeeper;
+
+class ZookeeperStoreTest extends AbstractStoreTest
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function setupBeforeClass()
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        try {
+            $zookeeper = new Zookeeper(implode(',', array($zookeeper_server)));
+            // The session creation happens asynchronously, so we can wait for a second to make sure it is established.
+            sleep(1);
+            if (Zookeeper::CONNECTED_STATE !== $zookeeper->getState()) {
+                self::markTestSkipped('Unable to connect to the zookeeper host.');
+            }
+        } catch (Throwable $th) {
+            self::markTestSkipped('Unable to connect to the zookeeper host: '.$th->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return \Symfony\Component\Lock\Store\ZookeeperStore
+     */
+    public function getStore(): ZookeeperStore
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        $zookeeper = new Zookeeper(implode(',', array($zookeeper_server)));
+
+        return new ZookeeperStore($zookeeper);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Lock\Exception\LockConflictedException
```

Comment:
Please use `$this->expectException(LockConflictedException::class)` right before it's supposed to be thrown. Right now if `getStore()` throws it, this test would pass, while we probably expect it on the second `save($key)`

Reply:
This should be fine as is. If `getStore` throws an then it would fail saying that `Failed asserting that exception of type "WhateverException" matches expected exception "\Symfony\Component\Lock\Exception\LockConflictedException"`. `getStore` will never throw a `LockConflictedException`.

Reply:
"getStore will never throw a LockConflictedException", not in this pull request. Tests should be robust, thus if someone throws an exception in another PR at an earlier point and this test does not fail, the test is wrong.

Reply:
Removed this test as per @jderusse 's suggestion that I shouldn't be modifying the behavior from the parent class for this.